src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -160,5 +223,69 @@ public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
 
         /// <summary>Gets whether the system is case-sensitive.</summary>
         internal static bool IsCaseSensitive { get { return false; } }
+
+
+        /// <summary>
+        /// Returns the volume name for dos, UNC and device paths.
+        /// </summary>
+        internal static ReadOnlySpan<char> GetVolumeName(ReadOnlySpan<char> path)
+        {
+            if (!IsPathFullyQualified(path))
```

Comment:
As mentioned, this probably isn't correct for `c:`.  Just calling `GetPathRoot()` and bailing if it is empty may give you the desired behavior.

Reply:
``` 
if (!IsPathFullyQualified(path))
      return getPathroot(path)
```
do you mean like this ?

Reply:
No,

``` C#
var root = GetPathRoot(path);
if (root.Length == 0)
    return root;
```

Reply:
Remove 

``` C#
if (!IsPathFullyQualified(path))
      return getPathroot(path)
```